src/zarr/metadata.py

```diff
@@ -184,14 +181,15 @@ def dtype(self) -> np.dtype[Any]:
     def ndim(self) -> int:
         return len(self.shape)
 
-    def get_chunk_spec(self, _chunk_coords: ChunkCoords) -> ArraySpec:
+    def get_chunk_spec(self, _chunk_coords: ChunkCoords, order: Literal["C", "F"]) -> ArraySpec:
         assert isinstance(
             self.chunk_grid, RegularChunkGrid
         ), "Currently, only regular chunk grid is supported"
         return ArraySpec(
             shape=self.chunk_grid.chunk_shape,
             dtype=self.dtype,
             fill_value=self.fill_value,
+            order=order,
```

Comment:
this `ArraySpec` object ends up with the output order attached. And this is what we use throughout the codec pipelines, etc.